### PR TITLE
8269528: VectorAPI Long512VectorTest fails on X86 KNL target

### PR DESCRIPTION
**Pull request by jatin-bhateja**
Long512VectorTest shows assertion failure on fastdebug build and incorrectness issue in release build.

Problem is seen on targets missing AVX512BW feature, load_vector_mask operation loads the mask boolean array into a vector followed by signed expansion operation to match the vector width of the operation. 

load_vector_mask macro assembly routine uses a vpsubb instruction which necessities existence of AVX512BW feature while operating over 64 byte operands.

Load mask over Byte512 and Short512 species does not get intrinsified since Matcher::match_rule_supported_vector
check for legality of vector size against the maximum vector size supported by the target, in case of 64 byte subword type operation, an AVX512 target must support AVX512BW feature else vector size is reduced to 32 bytes. 

For a load mask operation over all other species i.e. Int512,Float512,Double512,Long512 underlined mask boolean array is 16 bytes or less, thus a 256bit AVX2 byte vpsubb should suffice, also subsequent unpacking instruction (VPMOVSD, VPMOVSQ) needs AVX512F feature.

legVec register class constrains the allocation set of operands of new loadmask instruction pattern for non-AVX512BW targets to lower register bank ([X/Y]MM0-15 registers).

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8269528](https://bugs.openjdk.java.net/browse/JDK-8269528): VectorAPI Long512VectorTest fails on X86 KNL target

### Reviewers
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)
 * [Sandhya Viswanathan](https://openjdk.java.net/census#sviswanathan) (@sviswa7 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk17 pull/174/head:pull/174` \
`$ git checkout pull/174`

Update a local copy of the PR: \
`$ git checkout pull/174` \
`$ git pull https://git.openjdk.java.net/jdk17 pull/174/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 174`

View PR using the GUI difftool: \
`$ git pr show -t 174`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk17/pull/174.diff">https://git.openjdk.java.net/jdk17/pull/174.diff</a>

</details>

### CODE REVIEW

**Review by vnkozlov — COMMENTED** [review]
Seems good.

What testing you did? I may need to test it before approval.

**Review by sviswa7 — APPROVED** [review]
Looks good to me. Thanks for fixing this issue.

**Comment by jatin-bhateja** [comment]
> Seems good.
> 
> What testing you did? I may need to test it before approval.

Hi Vladimir,
Long512VectorTest now passes on KNL platform. No explicit test added  for this PR

**Review by vnkozlov — APPROVED** [review]
tier1-3 testing passed.

**Pull request closed without merging**
